airflow/gcp/hooks/cloud_memorystore.py

```diff
@@ -0,0 +1,487 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Hooks for Cloud Memorystore service"""
+
+import json
+from typing import Dict, Sequence, Tuple, Union, Optional
+
+from google.api_core.exceptions import NotFound
+from google.api_core.retry import Retry
+from google.cloud.redis_v1.gapic.enums import FailoverInstanceRequest
+from google.cloud.redis_v1.types import FieldMask, InputConfig, Instance, OutputConfig
+from google.cloud.redis_v1 import CloudRedisClient
+from google.protobuf import json_format
+
+from airflow import AirflowException, version
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+
+
+class CloudMemorystoreHook(GoogleCloudBaseHook):
+    """
+    Hook for Google Cloud Memorystore APIs.
+
+    All the methods in the hook where project_id is used must be called with
+    keyword arguments rather than positional.
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    def __init__(self, gcp_conn_id: str = "google_cloud_default", delegate_to: str = None):
+        super().__init__(gcp_conn_id, delegate_to)
+        self._client = None  # type: Optional[CloudRedisClient]
+
+    def get_conn(self,):
+        """
+        Retrieves client library object that allow access to Cloud Memorystore service.
+
+        """
+        if not self._client:
+            self._client = CloudRedisClient(credentials=self._get_credentials())
+        return self._client
+
+    @staticmethod
+    def _dict_to_proto(py_dict: Dict, proto):
+        """
+        Converts a python dictionary to the proto supplied
+
+        :param py_dict: The dictionary to convert
+        :type py_dict: dict
+        :param proto: The proto object to merge with dictionary
+        :type proto: protobuf
+        :return: A parsed python dictionary in provided proto format
+        :raises:
+            ParseError: On JSON parsing problems.
+        """
+        dict_json_str = json.dumps(py_dict)
+        return json_format.Parse(dict_json_str, proto)
```

Comment:
Can't we use `json_format.ParseDict`? That's what I did in DTS
```python
new_config = {'key' : 'value'}
ParseDict(new_config, TransferConfig())
```

Reply:
Other hook e.g. ``GKEClusterHook`` use this method.

Reply:
You mean dumps + json_format.Parse? I am not sure if there's reason for custom method  if there is one already implemented.

Reply:
Fixed.